source/_references/stacks.md

```diff
@@ -356,19 +357,27 @@ In table form, with a 300px image:
 
 rokka does some advanced image size optimizations on your images by default. As the optimizations are not always fast, it does those in the background to not slow down your first render request to an image. Therefore the first request on a newly rendered image will not have those optimizations applied, but requests made 10-30 seconds later will have those applied.
 
-For JPEGs you can prevent that delay with the stack option `optim.immediate.jpeg`. rokka then does it right on the first render and not only a few seconds later. This will make your first render a little bit slower, but won't make a difference for later requests. This can also be very useful, if you want to see the final image during developing right away (eg. for deciding about the appropriate jpeg quality setting).
+You can prevent that delay with the stack option `optim.immediate`. rokka then does it right on the first render and not only a few seconds later asynchronously. This will make your first render slower, but won't make a difference for later requests. This can also be very useful, if you want to see the final image during developing right away (eg. for deciding about the appropriate image quality setting). We recommend not to use this feature in production environments, since it can degrade your end user experience.
 
 For PNG and lossless WebP we use [pngquant](https://pngquant.org/) to make the image size significantly smaller as long as the quality doesn't degrade. We additionally compress PNG with [zopflipng](https://github.com/google/zopfli) to make them even smaller.
 
-For JPEG we recompress the lossless rendered image with [MozJPEG](https://github.com/mozilla/mozjpeg) and [jpeg-archive](https://github.com/danielgtaylor/jpeg-archive) and match the quality of the initially rendered image. This makes the image size approx. 75% - 90% the size of the initially rendered one, sometimes even less.
+For JPEG and lossy WebP images the approach choosen depends on your stack settings.
+
+By default or set `optim.quality` explicitly, rokka tries to find the best possible compression rate for your chosen `optim.quality`. It compresses the image several times (from the original image) with different quality settings and choses the one with the smallest filesize, which still looks good. This can lead to quite some savings in file size, but sometimes, depending on the image, it can produce a bigger image for a better perceived quality.
+
+We highly recommend to use this feature instead of the hard-wired compression settings with `jpeg.quality` and `webp.quality`.
+
+If you you used `jpg.quality` or `webp.quality` in your stack settings, rokka still tries to recompress the image to a lower filesize while keeping your set quality, but won't do much image depending and adaptive adjustments.
+
+If use set `optim.quality` and `jpg.quality`, rokka uses the `jpg.quality` setting for the inital rendering and `optim.quality` for the later adaptive optimization. 
```

Comment:
> If use set `optim.quality` and `jpg.quality`,

Did you mean

> If you set `optim.quality` and `jpg.quality`,

or what was the intention?

Reply:
yes, fixed it